src/BootstrapMixin.js

```diff
@@ -1,11 +1,16 @@
 import React from 'react';
 import constants from './constants';
+import CustomPropTypes from './utils/CustomPropTypes';
 
 const BootstrapMixin = {
   propTypes: {
-    bsClass: React.PropTypes.oneOf(Object.keys(constants.CLASSES)),
-    bsStyle: React.PropTypes.oneOf(Object.keys(constants.STYLES)),
-    bsSize: React.PropTypes.oneOf(Object.keys(constants.SIZES))
+    bsClass: CustomPropTypes.keyOf(constants.CLASSES),
+    bsStyle: CustomPropTypes.keyOf(constants.STYLES),
+    bsSize: CustomPropTypes.keyOf(constants.SIZES)
+  },
+
+  addStyle(name) {
+    constants.STYLES[name] = name;
```

Comment:
Agreed, particularly since mixins aren't supported for ES6 classes, so I figure `BootstrapMixin` might disappear someday.